source/docs/software/advanced-control/state-space/state-space-flywheel-walkthrough.rst

```diff
@@ -1,4 +1,123 @@
 
+The goal of this tutorial is to provide "end-to-end" instruction on implementing a state-space controller on a flywheel.  By following this tutorial, readers will learn how to:
+
+1. Create an accurate state-space model of a flywheel using :term:`system identification` or CAD software
+2. Implement a Kalman Filter to laglessly filter encoder velocity measurements.
+3. Implement a :ref:`LQR <docs/software/advanced-control/state-space/state-space-intro:The Linear-Quadratic Regulator>` feedback controller which, when combined with model-based feedforward, will generate voltrage :term:input:s to drive the flywheel to a :term:`reference`.
+
+This tutorial is intended to be approachable for teams without a great deal of programming expertise.  While the WPILib library offers significant flexibility in the manner in which its state-space control features are implemented, closely following the implementation outlined in this tutorial should provide teams with a basic structure which can be reused for a variety of state-space systems.
+
+.. todo:: link to allwpilib example
+
+Why use state-space control?
+============================
+
+Because IIC is a meme award. Thank you for coming to my ted talk
+
+.. todo:: real explication.
+
+Modeling our flywheel
+=====================
+
+The idea of modeling the :term:`dynamics` of physical systems using a system of equations is central to state-space control and modern control theory. Recall that continuous state-space systems are modeled using the following system of equations:
+
+.. math::
+    \dot{\mathbf{x}} &= \mathbf{A}\mathbf{x} + \mathbf{B}\mathbf{u} \\
+    \mathbf{y} &= \mathbf{C}\mathbf{x} + \mathbf{D}\mathbf{u} 
+
+Where :term:`x-dot` is the rate of change of the :term:`system`'s :term:`state`, :math:`\mathbf{x}` is the system's current state, :math:`\mathbf{u}` is the :term:`input` to the system, and :math:`\mathbf{y}` is the sytem's :term:`output`.
+
+Let's use this system of equations to model our flywheel in two different ways. We'll first model it using :term:`system identification` using the frc-characterization toolsuite, and then model it based on the motor and flywheel's angular moment of inertia.
+
+The first step of building up our state-space system is picking our system's states. We can pick anything we want as a state -- we could pick completely unrelated states if we wanted -- but it helps to pick states that are important. We can include :term:`hidden state`\s in our state (such as elevator velocity if we were only able to measure its position) and let our Kalman Filter estimate their value if desired. Remember that the states we choose will be driven towards their :term:`reference` by our :ref:`Linear-Quadratic Regulator <docs/software/advanced-control/state-space/state-space-intro:The Linear-Quadratic Regulator>`.
```

Comment:
Replace "estimate their value" with "estimate their values" since you are referring to hidden states (plural).